src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -784,16 +802,21 @@ where
                     let has_errors = batch.source_errors.first();
                     let has_messages = batch.messages.values().any(|vs| !vs.is_empty());
 
-                    let maybe_health = match (has_errors, has_messages) {
-                        (Some(error), _) => {
+                    let maybe_health = match (has_errors, &batch.status_update, has_messages) {
+                        (Some(error), _, _) => {
                             // Arguably this case should be "failed", since generally a source
                             // cannot recover by the time an error reaches this far down the pipe.
                             // However, we don't actually shut down the source on error yet, so
                             // treating this as a possibly-temporary stall for now.
                             Some(HealthStatus::StalledWithError(error.error.to_string()))
                         }
-                        (_, true) => Some(HealthStatus::Running),
-                        (None, false) => None,
+                        (_, Some(status), _) => {
+                            // This is the transient error case, and is correctly represented as
+                            // a (temporary) stall.
+                            Some(status.clone())
+                        }
+                        (_, _, true) => Some(HealthStatus::Running),
+                        (None, _, false) => None,
```

Comment:
I believe we could fill these out as `(None, None, true)` and `(None, None, false)`?  I think this makes it a little easier to read through quicker

If you think I'm being too particular about these match statements, please feel free to ignore!